### PR TITLE
[HB-4992] Simplify SKAdNetwork Integration Process

### PR DESCRIPTION
**Pull request by SCastanedaMunoz**
# Description

Helium Unity should have had some sort of reference as to how to integrate SKAdNetwork, this PR exists to address that and make it easier for developers to have such capability enabled.

# TODO 

We might want to add some checks in case the webrequests fail or timeout. This is just the initial PR 👍🏻

### CODE REVIEW

File: com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            foreach (var skadnetworkID in ids)
+            {
+                var skanId = skanItems.AddDict();
+                skanId.SetString(SKAdNetworkIdKey, skadnetworkID);
+            }
+
+            plist.WriteToFile(plistPath);
+        }
+
+        private const string AppLovin = "https://dash.applovin.com/docs/v1/skadnetwork_ids.json";
```
Comment (cb-jpadilla) [inline com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs:52]: Looks like there are some partner SDKs missing. They should have a json/xml link, right? 🤔

File: com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        private const string Chartboost = "https://a3.chartboost.com/skadnetworkids.json";
+        private const string Fyber = "https://fyber-engineering.github.io/SKAdNetworks/skadnetwork.json";
+        private const string InMobi = "https://www.inmobi.com/skadnetworkids.json";
+        private const string TapJoy = "https://skadnetwork.tapjoy.com/skadnetworkids.json";
+        private const string Vungle = "https://vungle.com/skadnetworkids.json";
+        private const string Unity = "https://skan.mz.unity3d.com/v3/partner/skadnetworks.plist.json";
+        private const string AdColony = "https://raw.githubusercontent.com/AdColony/AdColony-iOS-SDK/master/AdNetworks.csv";
```
Comment (cb-jpadilla) [inline com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs:59]: There's no json for AdColony?? 😳

**Review by cb-jpadilla — COMMENTED** [review]
A few comments for now.

File: com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs (hunk 1 above)
Comment (cb-jpadilla) [reply com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs:52]: Aw.. rip. Yeah, after a while, there are repeats.

Comment (chauduyphanvu) [reply com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs:52]: ^ We will have a process to dedup them if needed.

File: com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+            void MergeIDs(params SKAdNetworkIds[] ids)
+            {
+                foreach (var id in ids)
+                {
+                    AppendToFinalList(id, ref idsToAdd);
+                }
+            }
+
+            // json compatible SkAdNetworkFetching
+            var appLovinIds = SKAdNetworkRequest(AppLovin);
```
Comment (cb-jpadilla) [inline com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs:83]: Would Amazon need to be added?

**Comment by cb-jpadilla** [comment]
Lgtm. Looking for these is something. Nicely done.

**Review by cb-jpadilla — APPROVED** [review]

**Review by kushG — APPROVED** [review]
lgtm

File: com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+            while (!request.isDone) { }
+
+            if (skanIdsRequest.error != null)
+            {
+                Debug.Log($"SKAdNetworkRequest failed with error: {skanIdsRequest.error}");
+                return new SKAdNetworkIds();
+            }
+
+            var skanIds = JsonUtility.FromJson<SKAdNetworkIds>(skanIdsRequest.downloadHandler.text);
```
Comment (chauduyphanvu) [inline com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs:183]: Nit: Do we need to try catch JSON parsing in Unity?

File: com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+        private static SKAdNetworkIds SkAdNetworkRequestUnity(string url) 
+        {
+            var skanIdsRequest = UnityWebRequest.Get(url);
+            var request = skanIdsRequest.SendWebRequest();
+
+            while (!request.isDone) { }
+            
+            var ret = new SKAdNetworkIds
+            {
+                company_name = "Unity",
```
Comment (chauduyphanvu) [inline com.chartboost.helium/Editor/HeliumSKAdNetworkIds.cs:196]: Always `Unity`?

**Review by chauduyphanvu — APPROVED** [review]

**Pull request merged**